File: api-node/healthcheck/healthcheck.ts
import http from "http";
import config from '../src/config/config';
import { IHealthOption } from "../src/types/interfaces";


const options: IHealthOption = {
    timeout: 2000,
    host: 'localhost',
    port: config.port || 3000,
    path: "/ping"
}


const request = http.request(options, (res) => {
        console.info("STATUS: " + res.statusCode);
        process.exitCode = res.statusCode === 200 ? 0 : 1;
        process.exit();
})


request.on("error", function (err) {
  console.error("ERROR", err);
  process.exit(1);
});

request.end();

File: api-flask/healthcheck/healthcheck.py
import http.client
import os
import sys

timeout = 2
host = 'localhost'
port = os.getenv('PORT', 5000)
path = "/ping"


def initiate_health_check():
    try:
        conn = http.client.HTTPConnection(host, port, timeout=timeout)
        conn.request("GET", path)
        res = conn.getresponse()
        print(f"STATUS: {res.status}")

        sys.exit(0 if res.status == 200 else 1)
    except Exception as err:
        print(f"ERROR: {err}")
    finally:
        if 'conn' in locals():
            conn.close()


initiate_health_check()